### PR TITLE
fix: Migrate to new Quarkus Dev UI

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**

### CODE REVIEW

File: quarkus-integration/quarkus/devui-integration-test/pom.xml [hunk 1]
```diff
@@ -18,7 +18,7 @@
     <!-- Code of integration tests should not be a part of test coverage reports. -->
     <sonar.coverage.exclusions>**/*</sonar.coverage.exclusions>
     <!-- Quarkus 2 and 3 use different context roots. -->
-    <dev.ui.root>/q/dev-v1</dev.ui.root>
+    <dev.ui.root>/q/dev-ui</dev.ui.root>
```
Comment (triceo) [inline quarkus-integration/quarkus/devui-integration-test/pom.xml:21]: Since `main` only uses Quarkus 3, I think we can go back to hardcoding this and removing the comment.

File: build/build-parent/pom.xml [hunk 2]
```diff
@@ -651,7 +651,8 @@
     <profile>
       <id>sonarcloud-analysis</id>
       <properties>
-        <sonar.coverage.jacoco.xmlReportPaths>${project.reporting.outputDirectory}/jacoco/jacoco.xml</sonar.coverage.jacoco.xmlReportPaths>
+        <sonar.coverage.jacoco.xmlReportPaths>${project.reporting.outputDirectory}/jacoco/jacoco.xml
+        </sonar.coverage.jacoco.xmlReportPaths>
```
Comment (triceo) [inline build/build-parent/pom.xml:655]: Unrelated change?

File: quarkus-integration/quarkus/deployment/src/main/java/ai/timefold/solver/quarkus/deployment/TimefoldProcessor.java [hunk 3]
```diff
@@ -148,20 +152,48 @@ DetermineIfNativeBuildItem ifNativeBuild() {
...
+                .title("Constraints")
+                .icon("font-awesome-solid:wrench")
+                .componentLink("constraints-component.js"));
+
+        return cardPageBuildItem;
+    }
+
+    @BuildStep(onlyIf = IsDevelopment.class)
+    public JsonRPCProvidersBuildItem registerRPCService() {
+        return new JsonRPCProvidersBuildItem("Timefold", TimefoldDevUIPropertiesRPCService.class);
```
Comment (triceo) [inline quarkus-integration/quarkus/deployment/src/main/java/ai/timefold/solver/quarkus/deployment/TimefoldProcessor.java:196]: Shall we make it "Timefold Solver"? That way, it'd be consistent with what the project is actually called.

File: quarkus-integration/quarkus/deployment/src/main/resources/dev-ui/config-component.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import {JsonRpc} from 'jsonrpc';
+import '@vaadin/icon';
+import '@vaadin/button';
+import {until} from 'lit/directives/until.js';
+import '@vaadin/grid';
+import '@vaadin/grid/vaadin-grid-sort-column.js';
+
+export class ConfigComponent extends LitElement {
+
+    jsonRpc = new JsonRpc("Timefold");
```
Comment (triceo) [inline quarkus-integration/quarkus/deployment/src/main/resources/dev-ui/config-component.js:11]: Timefold Solver?

File: quarkus-integration/quarkus/deployment/src/main/resources/dev-ui/constraints-component.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import '@vaadin/icon';
+import '@vaadin/button';
+import {until} from 'lit/directives/until.js';
+import '@vaadin/grid';
+import {columnBodyRenderer} from '@vaadin/grid/lit.js';
+import '@vaadin/grid/vaadin-grid-sort-column.js';
+
+export class ConstraintsComponent extends LitElement {
+
+    jsonRpc = new JsonRpc("Timefold");
```
Comment (triceo) [inline quarkus-integration/quarkus/deployment/src/main/resources/dev-ui/constraints-component.js:12]: Dtto.

File: quarkus-integration/quarkus/deployment/src/main/resources/dev-ui/model-component.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+import '@vaadin/icon';
+import '@vaadin/button';
+import {until} from 'lit/directives/until.js';
+import '@vaadin/grid';
+import {columnBodyRenderer} from '@vaadin/grid/lit.js';
+import '@vaadin/grid/vaadin-grid-sort-column.js';
+
+export class ModelComponent extends LitElement {
+
+    jsonRpc = new JsonRpc("Timefold");
```
Comment (triceo) [inline quarkus-integration/quarkus/deployment/src/main/resources/dev-ui/model-component.js:12]: Dtto.

File: quarkus-integration/quarkus/devui-integration-test/src/test/java/ai/timefold/solver/quarkus/it/devui/TimefoldDevUITest.java [hunk 7]
```diff
@@ -57,51 +46,28 @@ void testSolverConfigPage() throws ParserConfigurationException, SAXException, I
...
-        assertThat(model)
-                .contains("value=[Entity: ai.timefold.solver.quarkus.it.devui.domain.TestdataStringLengthShadowEntity]");
-        assertThat(model).contains(
-                "value=[Genuine Variables]]]]]], tbody[attributes={}; value=[tr[attributes={}; value=[td[attributes={colspan=1, rowspan=1}; value=[value]]");
-        assertThat(model).contains(
-                "value=[Shadow Variables]]]]]], tbody[attributes={}; value=[tr[attributes={}; value=[td[attributes={colspan=1, rowspan=1}; value=[length]]");
+    void testModelPage() throws Exception {
+        JsonNode modelResponse = super.executeJsonRPCMethod("getModelInfo");
+        assertThat(modelResponse.get("solutionClass").asText())
+                .contains("ai.timefold.solver.quarkus.it.devui.domain.TestdataStringLengthShadowSolution");
```
Comment (triceo) [inline quarkus-integration/quarkus/devui-integration-test/src/test/java/ai/timefold/solver/quarkus/it/devui/TimefoldDevUITest.java:52]: I'd use the `class.getCanonicalName()` throughout this test, to make this safe for refactoring.

**Review by triceo — APPROVED** [review]
LGTM with some changes.

File: build/build-parent/pom.xml (hunk 2 above)
Comment (Christopher-Chianelli) [reply build/build-parent/pom.xml:655]: Unrelated change; from IDEA XML formatter. (/me learns one can use "format only changed lines" instead of "format whole file" as a save action).

**Comment by triceo** [comment]
Thank you, Chris!
